assets/src/edit-story/components/library/panes/text/textSets/textSetsPane.js

```diff
@@ -0,0 +1,185 @@
+} from '../../../../../utils/getInUseFonts';
+import { Container as SectionContainer } from '../../../common/section';
+import { virtualPaneContainer } from '../../shared/virtualizedPanelGrid';
+import TextSets from './textSets';
+import { CATEGORIES, PANE_TEXT } from './constants';
+
+const TitleBar = styled.div`
+  display: flex;
+  justify-content: space-between;
+  padding: 20px 0;
+`;
+
+const TextSetsToggle = styled.div`
+  display: flex;
+  p {
+    margin: auto 12px;
+  }
+`;
+
+const TextSetsWrapper = styled.div`
+  ${virtualPaneContainer};
+`;
+
+function TextSetsPane({ paneRef }) {
+  const { textSets } = useLibrary(({ state: { textSets } }) => ({ textSets }));
+  const [showInUse, setShowInUse] = useState(false);
+
+  const allTextSets = useMemo(() => Object.values(textSets).flat(), [textSets]);
+  const storyPages = useStory(({ state: { pages } }) => pages);
+
+  const [selectedCat, setSelectedCat] = useState(
+    localStore.getItemByKey(`${LOCAL_STORAGE_PREFIX.TEXT_SET_SETTINGS}`)
+      ?.selectedCategory
+  );
+
+  const getTextSetsForInUseFonts = useCallback(
+    () =>
+      getTextSetsForFonts({
+        fonts: getInUseFontsForPages(storyPages),
+        textSets: selectedCat ? textSets[selectedCat] : allTextSets,
+      }),
+    [allTextSets, storyPages, selectedCat, textSets]
+  );
+
+  const addIdToTextSets = useCallback(
+    (sets) =>
+      sets.map((elements) => {
+        return {
+          id: `text_set_${uuidv4()}`,
+          elements,
+        };
+      }),
+    []
+  );
+  const filteredTextSets = useMemo(() => {
+    if (showInUse) {
+      return addIdToTextSets(getTextSetsForInUseFonts());
+    }
+    return addIdToTextSets(selectedCat ? textSets[selectedCat] : allTextSets);
+  }, [
+    selectedCat,
+    textSets,
+    allTextSets,
+    addIdToTextSets,
+    getTextSetsForInUseFonts,
+    showInUse,
+  ]);
+
+  const categories = useMemo(
+    () => [
+      ...Object.keys(textSets).map((cat) => ({
+        id: cat,
+        label: CATEGORIES[cat] ?? cat,
+      })),
+    ],
+    [textSets]
+  );
+
+  const handleSelectedCategory = useCallback((selectedCategory) => {
+    setSelectedCat(selectedCategory);
+    localStore.setItemByKey(`${LOCAL_STORAGE_PREFIX.TEXT_SET_SETTINGS}`, {
+      selectedCategory,
+    });
+  }, []);
+
+  const onChangeShowInUse = useCallback(
+    () => requestAnimationFrame(() => setShowInUse((prevVal) => !prevVal)),
+    [setShowInUse]
+  );
+
+  useEffect(() => {
+    trackEvent('search', {
+      search_type: 'textsets',
+      search_term: '',
+      search_category: selectedCat,
+      search_filter: showInUse ? 'show_in_use' : undefined,
+    });
+  }, [selectedCat, showInUse]);
+
+  const sectionId = useMemo(() => `section-${uuidv4()}`, []);
+  const toggleId = useMemo(() => `toggle_text_sets_${uuidv4()}`, []);
+  return (
+    <SectionContainer id={sectionId}>
+      <TitleBar>
+        <Headline size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.XX_SMALL}>
```

Comment:
Getting a proptype error here:
![image](https://user-images.githubusercontent.com/22185279/109525035-5b331e00-7a6e-11eb-9407-0dff3b4850ae.png)

Maybe to stay consistent with the `<Text />` api we could add a default

Reply:
ah that was my bad, i made `as` mandatory in the headline so that you had pick where it went in the heading outline of the output html. 😂 so here i just forgot.